specs/unit/services/events.spec.ts

```diff
@@ -1,99 +1,348 @@
+        ))
+      )
     })
 
     afterEach(async () => {
-      await pubsub.unsubscribe(subId)
+      await Promise.all(subIds.map(id => pubsub.unsubscribe(id)))
     })
 
-    it('Fires a message created event when a message is created', async () => {
-      expect(events).to.be.of.length(0)
+    describe('Message events', () => {
 
-      const message = await factories.messageFactory.create({})
+      it('fires a MESSAGE created event when a message is created', async () => {
+        expect(events).to.be.of.length(0)
 
-      await waitFor(100);
+        const message = await factories.messageFactory.create({})
 
-      expect(events).to.be.of.length(1)
+        await waitFor(100);
 
-      const [ev, payload] = events[0];
+        expect(events).to.be.of.length(1)
 
-      expect(ev).to.eq(PubSubEvent.MESSAGE)
-      expect(payload).to.have.keys('action', 'message');
-      expect(payload.action).to.eq('create');
-      expect(payload.message.id).to.eq(message.id);
-      expect(payload.message.content).to.deep.equal(message.content);
-    })
+        const [ev, payload] = events[0];
+
+        expect(ev).to.eq(PubSubEvent.MESSAGE)
+        expect(payload).to.have.keys('action', 'message');
+        expect(payload.action).to.eq('create');
+        expect(payload.message.id).to.eq(message.id);
+        expect(payload.message.content).to.deep.equal(message.content);
+      })
+
+      it('fires a MESSAGE created event when a message is created via an upsert', async () => {
+        expect(events).to.be.of.length(0)
 
-    it('Fires a message created event when a message is created via an upsert', async () => {
-      expect(events).to.be.of.length(0)
+        const conversation = await factories.conversationFactory.create({});
+        const message = factories.messageFactory.build({ conversationId: conversation.id });
 
-      const conversation = await factories.conversationFactory.create({});
-      const message = factories.messageFactory.build({ conversationId: conversation.id });
+        await db.message.upsert({
+          where: { id: message.id },
+          create: message,
+          update: {}
+        })
 
-      await db.message.upsert({
-        where: { id: message.id },
-        create: message,
-        update: {}
+        await waitFor(100);
+
+        expect(events).to.be.of.length(1)
+
+        const [ev, payload] = events[0];
+
+        expect(ev).to.eq(PubSubEvent.MESSAGE)
+        expect(payload).to.have.keys('action', 'message');
+        expect(payload.action).to.eq('create');
+        expect(payload.message.id).to.eq(message.id);
+        expect(payload.message.content).to.deep.equal(message.content);
       })
 
-      await waitFor(100);
+      it('fires multiple MESSAGE updated events for batch updateMany operations', async () => {
+        expect(events).to.be.of.length(0)
+
+        const messages = await factories.messageFactory.createList(3);
+
+        await waitFor(100);
```

Comment:
Just curious if it is possible to wait for a certain outcome to occur instead. Sort of like the [waitFor](https://testing-library.com/docs/dom-testing-library/api-async/#waitfor) provided by testing library?

Reply:
for sure, I plan on improving this. Already did on the gql subscription tests